bin.src/convert_refcat_to_nJy.py

```diff
@@ -0,0 +1,128 @@
+import lsst.log
+
+
+def is_old_schema(filename):
+    """Check whether this file's schema has "old-style" fluxes."""
+    catalog = lsst.afw.table.SimpleCatalog.readFits(filename)
+    return not hasNanojanskyFluxUnits(catalog.schema)
+
+
+def process_one(filename, write=False, quiet=False):
+    """Convert one file in-place from Jy (or no units) to nJy fluxes.
+
+    Parameters
+    ----------
+    filename : `str`
+        The file to convert.
+    write : `bool`, optional
+        Write the converted catalog out, overwriting the read in catalog?
+    quiet : `bool`, optional
+        Do not print messages about files read/written or fields found?
+    """
+    log = lsst.log.Log()
+    if quiet:
+        log.setLevel(lsst.log.WARN)
+
+    log.info(f"Reading: {filename}")
+    catalog = lsst.afw.table.SimpleCatalog.readFits(filename)
+
+    output = convertToNanojansky(catalog, log, doConvert=write)
+
+    if write:
+        output.writeFits(filename)
+        log.info(f"Wrote: {filename}")
+
+
+def main():
+    import argparse
+    import sys
+
+    class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter, argparse.RawDescriptionHelpFormatter):
+        pass
+
+    parser = argparse.ArgumentParser(description=__doc__,
+                                     formatter_class=CustomFormatter)
+    parser.add_argument("path",
+                        help="Directory containing the reference catalogs to overwrite."
+                        " All files with a `.fits` extension in the directory will be processed.")
+    parser.add_argument('-n', '--nprocesses', default=1, type=int,
+                        help="Number of processes to use when reading and writing files.")
+    parser.add_argument('--write', action="store_true",
+                        help="Write the corrected files (default just prints what would have changed).")
+    parser.add_argument('--quiet', action="store_true",
+                        help="Be less verbose about what files and fields are being converted.")
+    args = parser.parse_args()
+
+    schema_file = os.path.join(args.path, "master_schema.fits")
+    if not os.path.isfile(schema_file):
+        print("Error: Cannot find master_schema.fits in supplied path:", args.path)
+        sys.exit(-1)
+    if not is_old_schema(schema_file):
+        print("Catalog does not contain old-style fluxes; nothing to convert.")
+        sys.exit(0)
+
+    files = glob.glob(os.path.join(args.path, "*.fits"))
+    with concurrent.futures.ProcessPoolExecutor(max_workers=args.nprocesses) as executor:
+        futures = executor.map(process_one, files, itertools.repeat(args.write), itertools.repeat(args.quiet))
+        # so that exceptions don't get lost
+        for future in futures:
+            pass
+
+    if args.write:
+        with open(os.path.join(args.path, 'config.py'), 'a') as configFile:
+            configFile.write("\n\n# Updated refcat from version 0 to have nJy flux units\n")
```

Comment:
There is no such thing as `format_version=0` in any existing files and there never will be since I introduced that version in this ticket, which also makes version 1 the default (with no way to write version 0).

I guess we could make an explicit check of `format_version` part of `is_old_schema`, in addition to the check of types? How does one read a config file like this outside of the `pex.Config` environment?

Reply:
I understand this may be more complicated than you want, but the safest thing is to load the config, set the version and write it back out.  This is also presumably future proof.

(untested) code snippet follows:
```
import lsst.meas.algorithms as meas_alg
config = meas_alg.ingestIndexReferenceTask.DatasetConfig()
config.load(os.path.join(args.path, 'config.py'))
config.format_version = 1
config.save(os.path.join(args.path, 'config.py'))
```